src/Command/Helper/TreeHelper.php

```diff
@@ -0,0 +1,89 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP Project
+ * @since         5.3.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Command\Helper;
+
+use Cake\Console\Helper;
+
+/**
+ * Tree command helper.
+ *
+ * Formats nested arrays into a tree with dashes and pipes.
+ */
+class TreeHelper extends Helper
+{
+    /**
+     * @inheritDoc
+     */
+    public function output(array $args): void
+    {
+        $isList = array_is_list($args);
+        foreach ($args as $key => $value) {
+            if (is_array($value)) {
+                $this->_io->out((string)$key);
+                $this->outputArray($value, '');
+            } elseif (!$isList) {
+                $this->_io->out((string)$key);
+                $this->outputValue($value, '└── ');
+            } else {
+                $this->outputValue($value, '');
+            }
+        }
+    }
+
+    /**
+     * Output an array in a tree.
+     *
+     * @param array $array
+     * @param string $prefix
+     * @return void
+     */
+    protected function outputArray(array $array, string $prefix): void
+    {
+        $i = 1;
+        $isList = array_is_list($array);
+        foreach ($array as $key => $value) {
+            $isLast = $i++ === count($array);
```

Comment:
The `count()` call should be moved out of the loop.

Reply:
True, but I had hoped it was optimized to a single value lookup anyway.